cmd/vic-machine/validate/validator.go

```diff
@@ -119,7 +122,64 @@ func (v *Validator) GetResourcePool(input *data.Data) (*object.ResourcePool, err
 	return v.Session.Pool, nil
 }
 
-func (v *Validator) getConnection(input *data.Data) (bool, error) {
+func (v *Validator) GetVCH(input *data.Data) (*object.ResourcePool, error) {
+	rp, err := v.GetResourcePool(input)
+	if err != nil {
+		return nil, err
+	}
+	vchPath := fmt.Sprintf("%s/%s", rp.InventoryPath, input.DisplayName)
+	vchPool, err := v.Session.Finder.ResourcePool(v.Context, vchPath)
+	if err != nil {
+		log.Errorf("Failed to get VCH resource pool %s: %s", vchPath, err)
+		return nil, err
+	}
+	return vchPool, nil
+}
+
+func (v *Validator) GetVCHConfig(vch *object.ResourcePool, name string) (*metadata.VirtualContainerHostConfigSpec, error) {
+	var err error
+	var vms []*vm.VirtualMachine
+
+	rp := compute.NewResourcePool(v.Context, v.Session, vch.Reference())
+	if vms, err = rp.GetChildrenVMs(v.Context, v.Session); err != nil {
+		log.Errorf("Failed to get VCH children, %s", err)
+		return nil, err
+	}
+	var vmName string
+	for _, vm := range vms {
+		if vmName, err = vm.LoadName(v.Context); err != nil {
+			err = errors.Errorf("Failed to get VM name of %s, %s", vm.Reference(), err)
+			log.Errorf("%s", err)
+			return nil, err
+		}
+		if vmName == name {
+			//this is the appliance vm
+			mapConfig, err := vm.FetchExtraConfig(v.Context)
+			if err != nil {
+				err = errors.Errorf("Failed to get VM extra config of %s, %s", vmName, err)
+				log.Errorf("%s", err)
+				return nil, err
+			}
+			data := extraconfig.MapSource(mapConfig)
+			vchConfig := &metadata.VirtualContainerHostConfigSpec{}
+			result := extraconfig.DecodeWithPrefix(data, vchConfig, "guestinfo.vch")
```

Comment:
You'll need to update this to simply `Decode(...)` if it's merged after my PR.
We should also check that vchConfig.ID contains the moref of the VM - it's a good additional check that this is a VCH appliance.